py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,292 @@
+
+
+def parse_response_scope(scope: str) -> Dict[Any, Any]:
+    """
+    Parses scope string of the response into a dict
+    """
+    param_dict = {}
+    parts = scope.split(",")
+    for part in parts:
+        if ":" in part:
+            sub_parts = part.split(":")
+            if len(sub_parts) == 2:
+                param_dict[sub_parts[0].strip()] = sub_parts[1].strip()
+
+    return param_dict
+
+
+def parse_and_post(
+    series_list: List[DatadogResponseSeries],
+    parsed_unit: UsageUnit,
+    usage_accumulator: UsageAccumulator,
+) -> None:
+    """
+    Iterates through the series_list and
+    posts the data points to UsageAccountant (hence, Kafka).
+    Sample response:
+
+    {
+       "status":"ok",
+       "res_type":"time_series",
+       "resp_version":1,
+       "query":"avg: redis.mem.peak{app_feature: shared}
+       by{shared_resource_id}.rollup(5)",
+       "from_date":1721083881000,
+       "to_date":1721083891000,
+       "series":[
+          {
+             "unit":[
+                {
+                   "family":"bytes",
+                   "id":2,
+                   "name":"byte",
+                   "short_name":"B",
+                   "plural":"bytes",
+                   "scale_factor":1.0
+                },
+                None
+             ],
+             "query_index":0,
+             "aggr":"avg",
+             "metric":"redis.mem.peak",
+             "tag_set":["shared_resource_id: rc_long_redis"],
+             "expression":"avg: redis.mem.peak{shared_resource_id:
+             rc_long_redis,app_feature: shared}.rollup(5)",
+             "scope":"app_feature: shared, shared_resource_id: rc_long_redis",
+             "interval":5,
+             "length":2,
+             "start":1721083885000,
+             "end":1721083894000,
+             "pointlist":[
+                [1721083885000.0, 2494386918.6666665],
+                [1721083890000.0, 2494386918.6666665]
+             ],
+             "display_name":"redis.mem.peak",
+             "attributes":{}
+          }
+       ],
+       "values":[],
+       "times":[],
+       "message":"",
+       "group_by":["shared_resource_id"]
+    }
+
+    series_list: List of series dict extracted from API response
+    parsed_unit: UsageUnit parsed from API response
+    usage_accumulated: Object of UsageAccumulator class
+    """
+    for series in series_list:
+        scope = series.get("scope", "")
+        scope_dict = parse_response_scope(scope)
+        if (
+            "shared_resource_id" not in scope_dict.keys()
+            or "app_feature" not in scope_dict.keys()
+        ):
+            exception_msg = (
+                f"Required parameters, shared_resource_id and/or app_feature "
+                f"not found in series.scope of the series {series} received."
+            )
+            raise Exception(exception_msg)
+
+        resource = scope_dict.get("shared_resource_id", "")
+        app_feature = scope_dict.get("app_feature", "")
```

Comment:
You have already checked that these scopes are in the dict.
Just ensure this will never return empty string:
```
resource = scope_dict["shared_resource_id"]
app_feature = scope_dict["app_feature"]
```